server/server_test.go

```diff
@@ -347,6 +347,7 @@ func TestWatcherError(t *testing.T) {
 	srv := NewServer(watcher, poster, fileGetter, analyzers, &store.NoopEventOperator{}, &store.NoopCommentOperator{})
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	ctx = ctxlog.Set(ctx, logMock)
```

Comment:
It looks like `Set` is only used in this test. Why is it needed here? Can't you leave the previous `log.DefaultLogger = logMock`, and let `ctxlog.Get` return that for this `ctx`?

Reply:
We need it because this test actually checking log messages.

We are interested of the log messages of this test only and don't what to affect other tests.

Now when the logger is scoped it's better to set mock to particular test scope than playing with globals.

Another problem with using globals here: this test would rely on particular behaviour of `ctxlog.Get` (returning `log.DefaultLogger`) but it shouldn't.